tokens/batched/batched_test.go

```diff
@@ -0,0 +1,1015 @@
+package batched
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/binary"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/cloudflare/circl/oprf"
+	"github.com/cloudflare/pat-go/tokens"
+	"github.com/cloudflare/pat-go/tokens/type1"
+	"github.com/cloudflare/pat-go/tokens/type2"
+	"github.com/cloudflare/pat-go/tokens/typeF91A"
+	"github.com/cloudflare/pat-go/util"
+	"golang.org/x/crypto/hkdf"
+)
+
+// 2048-bit RSA private key
+// nosemgrep
+const testTokenPrivateKey = `
+-----BEGIN RSA PRIVATE KEY-----
```

Comment:
what makes special to this key?
can we convert it to a call to:
`key := rsa.GenerateKey(sha3.NewSHAKE128())`
thus, we can avoid having keys in clear.

Reply:
this is the key used in type2 test. I did not want to diverge from what was generated there, and this is exactly how it's used.

Reply:
then you have to convince semgrep about it.

Reply:
if we agree it's a test key (and I think we agree), it's fine to ignore semgrep. it's not a blocking CI check

Reply:
There is an example in stdlib on how to use a testing key.
https://pkg.go.dev/crypto/rsa@master#example-GenerateKey-TestKey

Reply:
ok. so the proposal is not to use `GenerateKey`, but rather to declare the key as `RSA TESTING KEY` instead of `RSA PRIVATE KEY` and use a replace function?